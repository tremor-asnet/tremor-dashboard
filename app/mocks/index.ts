export * from "./card";
export * from "./profile";
export * from "./project";
export * from "./charts";
export * from "./analytics";
export * from "./sales";
export * from "./table-list";
export * from "./orders";
export * from "./product";
