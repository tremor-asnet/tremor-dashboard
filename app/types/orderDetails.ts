export interface InvoiceHeaderProps {
  id: number;
  createdAt: string;
  orderCode: string;
}

export interface OrderSummaryProps {
  productPrice: number;
  delivery: number;
  taxes: number;
}
