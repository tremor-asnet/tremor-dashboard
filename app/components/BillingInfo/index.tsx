import { Bold, Card, Flex } from "@tremor/react";

// Types
import { BillingInfoProps } from "@/types";

const BillingInfo = ({
  owner_name,
  company_name,
  email,
  vat,
}: BillingInfoProps) => (
  <Card className="h-full bg-tremor-primary dark:bg-dark-tremor-primary p-0 border-none ring-0 p-6">
    <Bold className="dark:text-white">Billing Information</Bold>
    <div className="opacity-100 bg-greyish mt-6 rounded-lg p-6 dark:bg-dark-tremor-primary">
      <span className="text-primary text-tremor-default font-semibold dark:text-white">
        {owner_name}
      </span>
      <div className="mt-4">
        <Flex className="flex-row mb-2">
          <span className="text-tremor-label text-tremor-content dark:text-dark-romance">
            Company Name:
            <span className="text-primary font-semibold pl-2.5  dark:text-white">
              {company_name}
            </span>
          </span>
        </Flex>
        <Flex className="flex-row mb-2">
          <span className="text-tremor-label text-tremor-content dark:text-dark-romance">
            Email Address:
            <span className="text-primary font-semibold pl-2.5  dark:text-white">
              {email}
            </span>
          </span>
        </Flex>
        <Flex className="flex-row mb-2">
          <span className="text-tremor-label text-tremor-content dark:text-dark-romance">
            VAT Number:
            <span className="text-primary font-semibold pl-2.5  dark:text-white">
              {vat}
            </span>
          </span>
        </Flex>
      </div>
    </div>
  </Card>
);

export default BillingInfo;
