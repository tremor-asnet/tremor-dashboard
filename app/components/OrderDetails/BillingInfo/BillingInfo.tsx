import { Bold, Flex } from "@tremor/react";

// Types
import { BillingInfoData } from "@/types";

const BillingInfo = ({
  billingData,
}: {
  billingData: Partial<BillingInfoData>;
}) => {
  const { ownerName, companyName, email, vat } = billingData;

  const billingInfoData = [
    {
      id: 1,
      type: "companyName",
      title: "Company Name",
      value: companyName,
    },
    {
      id: 2,
      type: "email",
      title: "Email Address",
      value: email,
    },
    {
      id: 3,
      type: "vat",
      title: "VAT Number",
      value: vat,
    },
  ];

  const checkRenderBillValue = (type: string, value?: string) => {
    return type === "email" ? (
      <a
        href={`mailto: ${value}`}
        className="text-xs text-primary font-bold dark:text-white ml-2.5">
        {value}
      </a>
    ) : (
      <p className="text-xs text-primary font-bold dark:text-white ml-2.5">
        {value}
      </p>
    );
  };

  const renderBillInfoData = billingInfoData.map(
    ({ id, type, title, value }) => {
      return (
        <Flex key={id} className="content-start justify-start">
          <p className="text-secondary dark:text-dark-romance text-xs dark:text-dark-lighter">
            {`${title}:`}
          </p>
          {checkRenderBillValue(type, value)}
        </Flex>
      );
    },
  );

  return (
    <>
      <Flex
        flexDirection="col"
        alignItems="start"
        className="bg-greyish mt-4 rounded-lg p-6 dark:bg-dark-tremor-primary">
        <span className="text-primary text-tremor-default font-semibold dark:text-white">
          {ownerName}
        </span>
        <Flex className="flex-col mt-4 gap-3">{renderBillInfoData}</Flex>
      </Flex>
    </>
  );
};

export default BillingInfo;
