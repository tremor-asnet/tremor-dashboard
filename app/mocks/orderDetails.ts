import { TrackOrderProps } from "@/types";

export const mockInvoiceHeader = {
  id: 241342,
  createdAt: "2024-01-11T01:48:52Z",
  orderCode: "KF332",
};

export const mockOrderSummary = {
  productPrice: 90,
  delivery: 14,
  taxes: 1.95,
};

export const mockTrackOrder: TrackOrderProps = {
  id: "#1832412",
  generateOrderAt: "2024-01-11T02:48:52Z",
  deliveredAt: "2024-01-11T02:48:52Z",
  transmitedToCourierAt: "2024-01-11T02:48:52Z",
  generateOrderId: "2024-01-11T02:48:52Z",
};
