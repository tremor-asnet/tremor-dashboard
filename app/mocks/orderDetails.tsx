import { FaBell } from "react-icons/fa";
import { MdInventory } from "react-icons/md";
import { IoCart } from "react-icons/io5";
import { StepOrderProps } from "@/types/orderDetail";

export const mockInvoiceHeader = {
  id: 241342,
  createdAt: "2024-01-11T01:48:52Z",
  orderCode: "KF332",
};

export const mockTrackOrder: StepOrderProps[] = [
  {
    iconInfo: <FaBell />,
    titleInfo: "Order received",
  },
  {
    iconInfo: <MdInventory />,
    titleInfo: "Generate order id",
  },
  {
    iconInfo: <IoCart />,
    titleInfo: "Order transmited to courier",
  },
];
