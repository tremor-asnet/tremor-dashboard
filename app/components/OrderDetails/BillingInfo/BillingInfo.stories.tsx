// Libs
import type { Meta, StoryObj } from "@storybook/react";

// Components
import BillingInfo from "./BillingInfo";

// Mocks
import { mockBillingInfo } from "@/mocks/orderDetails";

const meta = {
  title: "Components/BillingInfo",
  tags: ["autodocs"],
  component: BillingInfo,
} as Meta<typeof BillingInfo>;

export default meta;

type Story = StoryObj<typeof meta>;

export const Default: Story = {
  render: () => <BillingInfo billingData={mockBillingInfo} />,
};
