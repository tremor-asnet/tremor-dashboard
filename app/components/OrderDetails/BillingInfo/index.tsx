import { Bold, Flex } from "@tremor/react";

// Types
import { BillingInfoProps } from "@/types/orderDetails";

const BillingInfo = (billingData: BillingInfoProps) => {
  const listTitle = ["Company Name", "Email Address", "VAT Number"];
  const { ownerName, ...billingInfo } = billingData;

  return (
    <div>
      <Bold className="dark:text-white">Billing Information</Bold>
      <Flex
        flexDirection="col"
        alignItems="start"
        className="bg-greyish mt-4 rounded-lg p-6 dark:bg-dark-tremor-primary">
        <span className="text-primary text-tremor-default font-semibold dark:text-white">
          {billingData.ownerName}
        </span>
        <Flex
          className="mt-4 gap-3"
          alignItems="baseline"
          justifyContent="start">
          <div>
            {listTitle.map(title => (
              <p className="mb-2 text-xs dark:text-dark-lighter" key={title}>
                {`${title}:`}
              </p>
            ))}
          </div>
          <div>
            {Object.keys(billingInfo).map(item => {
              const data =
                billingInfo[item as keyof Omit<BillingInfoProps, "ownerName">];
              return (
                <p
                  key={`${item}`}
                  className="mb-2 font-bold text-xs dark:text-white">
                  {data}
                </p>
              );
            })}
          </div>
        </Flex>
      </Flex>
    </div>
  );
};

export default BillingInfo;
