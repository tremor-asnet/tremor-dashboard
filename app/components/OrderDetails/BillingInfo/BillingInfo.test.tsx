import { render } from "@testing-library/react";

// Components
import BillingInfo from ".";

// Mocks
import { mockBillingInfo } from "@/mocks/orderDetails";

describe("Testing BillingInfo component", () => {
  it("Should match snapshot", () => {
    const { container } = render(<BillingInfo billingData={mockBillingInfo} />);
    expect(container).toMatchSnapshot();
  });
});
