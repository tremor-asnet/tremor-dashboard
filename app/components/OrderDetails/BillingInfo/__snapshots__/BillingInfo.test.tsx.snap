// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing BillingInfo component Should match snapshot 1`] = `
<div>
  <div>
    <b
      class="text-inherit font-bold dark:text-white"
    >
      Billing Information
    </b>
    <div
      class="tremor-Flex-root flex w-full flex-col justify-between items-start bg-greyish mt-4 rounded-lg p-6 dark:bg-dark-tremor-primary"
    >
      <span
        class="text-primary text-tremor-default font-semibold dark:text-white"
      >
        Oliver Liam
      </span>
      <div
        class="tremor-Flex-root flex w-full flex-row justify-start items-baseline mt-4 gap-3"
      >
        <div
          class="tremor-Flex-root flex w-full flex-col justify-between items-start max-w-[90px]"
        >
          <p
            class="mb-2 text-xs dark:text-dark-lighter"
          >
            Company Name:
          </p>
          <p
            class="mb-2 text-xs dark:text-dark-lighter"
          >
            Email Address:
          </p>
          <p
            class="mb-2 text-xs dark:text-dark-lighter"
          >
            VAT Number:
          </p>
        </div>
        <div
          class="tremor-Flex-root flex w-full flex-col justify-between items-start"
        >
          <p
            class="mb-2 font-bold text-xs dark:text-white"
          >
            Viking Burrito
          </p>
          <p
            class="mb-2 font-bold text-xs dark:text-white"
          >
            oliver@burrito.com
          </p>
          <p
            class="mb-2 font-bold text-xs dark:text-white"
          >
            FRB1235476
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
